src/sets.jl

```diff
@@ -193,17 +193,19 @@ they should be unique values. The *k*th element in the set corresponds to the
 for a description of SOS constraints and their potential uses.
 This is a shortcut for the `MathOptInterface.SOS1` set.
 """
-struct SOS1 <: AbstractVectorSet
-    weights::Vector{Float64}
-    function SOS1(weights::AbstractVector = Float64[])
-        return new(convert(Vector{Float64}, weights))
+struct SOS1{T} <: AbstractVectorSet
+    weights::Vector{T}
+    function SOS1{T}(weights::AbstractVector = T[]) where {T}
+        return new{T}(convert(Vector{T}, weights))
     end
 end
-function moi_set(set::SOS1, dim::Int)
+SOS1() = SOS1{Int}()
```

Comment:
Add a comment that `Int` is chosen as a placeholder, and the true type is converted when adding to the model.